web/profiles/custom/yalesites_profile/modules/custom/ys_templated_content/src/Importers/ZipFileImporter.php

```diff
@@ -0,0 +1,245 @@
+<?php
+
+namespace Drupal\ys_templated_content\Importers;
+
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\Core\Url;
+use Drupal\ys_templated_content\FileTypes\ZipFile;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+/**
+ * Takes a zip file and modifies it for new import.
+ */
+class ZipFileImporter {
+  use StringTranslationTrait;
+
+  /**
+   * The content sync helper.
+   *
+   * @var \Drupal\single_content_sync\ContentSyncHelperInterface
+   */
+  protected $contentSyncHelper;
+
+  /**
+   * The import manager.
+   *
+   * @var \Drupal\ys_templated_content\Managers\ImportManager
+   */
+  protected $importManager;
+
+  /**
+   * The template modifier.
+   *
+   * @var \Drupal\ys_templated_content\Modifiers\TemplateModifier
+   */
+  protected $templateModifier;
+
+  /**
+   * ZipFileImporter constructor.
+   *
+   * @param \Drupal\ys_templated_content\Managers\ImportManager $importManager
+   *   The import manager.
+   */
+  public function __construct($importManager) {
+    $this->importManager = $importManager;
+    $this->contentSyncHelper = $importManager->getContentSyncHelper();
+    $this->templateModifier = $importManager->getTemplateModifier();
```

Comment:
This is not something I normally see where you import a service via another service. I see that these different importers are used within the ImportManager. I think most of the time I see this setup as a Factory rather than calling these classes with the `new` keyword.

Everything you have is working. There is no need to refactor. Just sharing what is top of mind.

Reply:
No, these are the kind of things I'd like to know.  Thanks for sharing it.  I really do want to make this where someone more experienced in Drupal and PHP would recognize the correct logical pattern, so I'll keep this on my list of things to maybe address after some of the other ones here and see where it can get us.